inc/namespace.php

```diff
@@ -31,3 +33,42 @@ function on_plugins_loaded() {
 		require_once ROOT_DIR . '/vendor/johnbillion/query-monitor/query-monitor.php';
 	}
 }
+
+function qm_file_path_map( $map ) : array {
+	// Chassis and Local Server
+	if ( file_exists( '/etc/chassis-constants' ) ) {
+		$json_string = file_get_contents( '/etc/chassis-constants' );
+		$data = json_decode( $json_string, true );
+		if ( ! empty( $data['synced_folders']['/chassis'] ) ) {
+			$folder_path = $data['synced_folders']['/chassis'] . '/content/themes/base';
+			$map['/chassis/'] = $data['synced_folders']['/chassis'] . '/';
+		}
+	}
+
+	return $map;
+}
+
+function qm_file_link_format( $format ) : string {
+	$editor = 'phpstorm';
+	if ( defined( 'QM_LOCAL_EDITOR' ) ) {
+		$editor = QM_LOCAL_EDITOR;
+	}
+	return qm_file_link_editor_format( $format, $editor );
+}
+
+function qm_file_link_editor_format( $format, $editor = null ) : string {
```

Comment:
Needs a phpdoc block.

Reply:
Is there a reason to split these functions out here?

Reply:
I wanted to separate the handling of the filter from the generation of the protocol. It also means I can write unit tests for the second function and pass the varying editor options as a parameter rather than having to define and undefine a constant